password_security/controllers/main.py

```diff
@@ -44,6 +45,7 @@ def web_login(self, *args, **kw):
             request.params['password']
         )
         if not uid:
+            request.uid = old_uid
```

Comment:
If we skip old_uid, we can not access `ir.model.data` to get public user because of security, it requires uid to get data. So i thank we should place `request.uid = request.env.ref('base.public_user').id` after `request.uid = old_uid`